scripts/compose.py

```diff
@@ -1696,6 +1700,18 @@ def add_arguments(cls, parser):
             '--opbeans-dotnet-version',
             default=cls.DEFAULT_AGENT_VERSION,
         )
+        parser.add_argument(
```

Comment:
can this be moved into the `OpbeansService` class rather than copy/pasted into each subclass?  Certainly possibly I missed something

Reply:
That's a good point, there are a couple of classes which don't need those flags:
- OpbeansGo
- OpbeansDotnet

The above classes use other flags:
- Go. --opbeans-go-branch and --opbeans-go-repo
- Dotnet. --opbeans-dotnet-branch and --opbeans-dotnet-repo

For further details:
- https://github.com/elastic/apm-integration-testing/pull/556/files#diff-d0df5dd5f0019c3b878fde30f4b7341bR1773
- https://github.com/elastic/apm-integration-testing/pull/556/files#diff-d0df5dd5f0019c3b878fde30f4b7341bR1704

While the remaining opbeans use:
- --opbeans-LANGUAJE-image
- --opbeans-LANGUAJE-version

Except `rum-js` which uses a different implementation.

The main reason is about the implementation details:
- Go and Dotnet clone the opbean repo and for such, it's required to use different arguments, while the other opbeans do consume the docker images.

That's the reason I didn't declare those arguments in the OpbeansService class at first, or it's possible to do it and then remove them in the classes which don't use them?

Reply:
That's a great explanation and reason to implement this way.  I'm happy to leave it as implemented.